app/lib/search/token_index.dart

```diff
@@ -183,6 +184,33 @@ class ScorePool<K> {
   }
 }
 
+/// A reusable pool for [BitArray] instances to spare some memory allocation.
+class BitArrayPool<K> {
+  final int _length;
+  final _pool = <BitArray>[];
+
+  BitArrayPool(this._length);
```

Comment:
Doesn't matter too much, but I guess it could be generic, sharing implentation with ScorePool (if you pass a constructor).

```suggestion
class AllocationPool<T> {
  final _pool = <T>[];
  final T Function() constructor;

  AllocationPool(this.constructor);
```

Reply:
I've checked but with the ScorePool's variable initialization parameter, it requires a bit of refactoring first. I'll do that in a follow-up PR.